pkg/probe/firewall_ippool.go

```diff
@@ -0,0 +1,91 @@
+package probe
+
+import (
+	"log"
+
+	"github.com/bluecmd/fortigate_exporter/pkg/http"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type IpPool struct {
+	Name      string  `json:"name"`
+	IPTotal   int     `json:"natip_total"`
+	IPInUse   int     `json:"natip_in_use"`
+	Clients   int     `json:"clients"`
+	Available float64 `json:"available"`
+	Used      int     `json:"used"`
+	Total     int     `json:"total"`
+}
+
+type IpPoolResponse struct {
+	Results map[string]IpPool `json:"results"`
+	VDOM    string            `json:"vdom"`
+	Version string            `json:"version"`
+}
+
+func probeFirewallIpPool(c http.FortiHTTP, meta *TargetMetadata) ([]prometheus.Metric, bool) {
+	var (
+		mAvailable = prometheus.NewDesc(
+			"fortigate_ippool_available",
+			"Percentage available in ippool",
+			[]string{"vdom", "name"}, nil,
+		)
+	)
+	var (
+		mIpUsed = prometheus.NewDesc(
+			"fortigate_ippool_ip_used",
+			"Ip addresses in use in ippool",
+			[]string{"vdom", "name"}, nil,
+		)
+	)
+	var (
+		mIpTotal = prometheus.NewDesc(
+			"fortigate_ippool_ip_total",
+			"Ip addresses total in ippool",
+			[]string{"vdom", "name"}, nil,
+		)
+	)
+	var (
+		mClients = prometheus.NewDesc(
+			"fortigate_ippool_clients",
+			"Amount of clients using ippool",
+			[]string{"vdom", "name"}, nil,
+		)
+	)
+	var (
+		mUsed = prometheus.NewDesc(
+			"fortigate_ippool_used",
+			"Amount of items used in ippool",
+			[]string{"vdom", "name"}, nil,
+		)
+	)
+	var (
+		mTotal = prometheus.NewDesc(
+			"fortigate_ippool_total",
```

Comment:
Have a look at https://prometheus.io/docs/practices/naming/ - the `_total` suffix is used for counter values (not gauge values) that are strictly accumulating. I would propose something like `fortigate_ippool_total_ips` or `fortigate_ippool_total_count` or something like that to add the unit.

Reply:
This should be fixed now